code/modules/projectiles/guns/energy/modular.dm

```diff
@@ -0,0 +1,241 @@
+/obj/item/weapon/gun/energy/laser/prototype
+	name = "laser prototype"
+	desc = "A custom prototype laser weapon."
+	icon_state = "laser"
+	item_state = "laser"
+	fire_sound = 'sound/weapons/Laser.ogg'
+	slot_flags = SLOT_BELT|SLOT_BACK
+	w_class = 3
+	force = 10
+	origin_tech = list(TECH_COMBAT = 3, TECH_MAGNET = 2)
+	matter = list(DEFAULT_WALL_MATERIAL = 2000)
+	can_turret = 0
+	zoomdevicename = null
+	var/origin_chassis
+	var/gun_type
+	var/list/gun_mods
+	var/obj/item/laser_components/capacitor/capacitor
+	var/obj/item/laser_components/focusing_lens/focusing_lens
+	var/chargetime
+	var/is_charging
+	var/criticality = 1 //multiplier for the negative effects of capacitor failures. Not just limited to critical failures.
+	var/named = 0
+	var/described = 0
+
+/obj/item/weapon/gun/energy/laser/prototype/examine()
+	..()
+	if(usr.Adjacent(src))
```

Comment:
Call `. = ..(user, 1)` instead of `..()` on the above line and check `.` here instead.